sstables/mx/reader.cc

```diff
@@ -1764,8 +1766,22 @@ class mx_sstable_full_scan_reader : public mp_row_consumer_reader_mx {
              integrity_check integrity)
         : mp_row_consumer_reader_mx(std::move(schema), permit, std::move(sst))
         , _consumer(this, _schema, std::move(permit), _schema->full_slice(), std::move(trace_state), streamed_mutation::forwarding::no, _sst)
-        , _context(data_consume_rows<DataConsumeRowsContext>(*_schema, _sst, _consumer, integrity))
-        , _monitor(mon) {
+        , _monitor(mon)
+        , _integrity(integrity) {}
+private:
+    bool is_initialized() const {
+        return bool(_context);
+    }
+    future<> maybe_initialize() {
+        if (is_initialized()) {
+            co_return;
+        }
+        if (_integrity) {
+            // Caller must retain a reference to checksum component while in use by the stream.
+            _checksum = co_await _sst->read_checksum();
+            co_await _sst->read_digest();
```

Comment:
Why not make this conditional? Maybe caller (or a parallel scrub) already loaded the component. Or `read_checksum()` and `read_digest()` already has this optimization (no-op if already loaded)?

Reply:
Both functions are idempotent. They act as no-ops if the components have already been loaded.

Reply:
Great, that is actually better than having all their callers have to do this.